### PR TITLE
Add example for micro

### LINKED ISSUE #13: Add an example project for micro
Support micro framework and add an example project for micro - https://github.com/zeit/micro.

Language - You can use either JS or TS.  

**Note: All example projects should follow similar conventions and folder structure as the existing ones here https://github.com/leapfrogtechnology/async-store/tree/master/examples**

### PR DESCRIPTION
**Pull request by sumansta**
Add example for [micro](https://github.com/zeit/micro) framework.

Resolves #13

### CODE REVIEW

File: examples/micro-http-server-ts/src/service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * An example function making use of the request context
+ * set in the store asynchronously (with delay).
+ */
+export function doSomethingAsync() {
+  // Do something with the request with a delay.
+  logger.debug('Simulating delayed access');
+
+  setTimeout(() => {
+    const [a, b, sum] = [store.get('a'), store.get('b'), store.get('sum')];
```
Comment (cham11ng) [inline examples/micro-http-server-ts/src/service.ts]:
```suggestion
	const res = store.getAll();
```

Comment (cham11ng) [reply examples/micro-http-server-ts/src/service.ts]: We have `getAll` function to fetch all the values of the store.

File: examples/micro-http-server-ts/src/service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    logger.info('Store contents: ' + JSON.stringify({ a, b, sum }));
```
Comment (cham11ng) [inline examples/micro-http-server-ts/src/service.ts]:
```suggestion
    logger.info('Store contents: ', data);
```

File: examples/micro-http-server-ts/src/service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * An example function making use of the request context
+ * set in the store asynchronously (with delay).
+ */
+export function doSomethingAsync() {
+  // Do something with the request with a delay.
+  logger.debug('Simulating delayed access');
+
+  setTimeout(() => {
```
Comment (cham11ng) [inline examples/micro-http-server-ts/src/service.ts]:
```ts
return new Promise((resolve, reject) => { 
// Do async task here.
})
```

File: examples/micro-http-server-ts/src/middlewares.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import * as store from '@leapfrogtechnology/async-store';
+
+import * as logger from './logger';
+import { doSomethingAsync } from './service';
+
+/**
+ * Middleware to set query params `a` and `b` on async-store.
+ *
+ * @param {any} query
+ *
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/middlewares.ts]: .

File: examples/micro-http-server-ts/package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  "devDependencies": {
+    "@types/qs": "^6.5.3",
+    "backpack-core": "^0.8.4",
+    "fork-ts-checker-webpack-plugin": "^1.5.0",
+    "ts-loader": "^6.1.2",
+    "tslint": "^5.20.0",
+    "tslint-config-leapfrog": "^1.0.3",
+    "tslint-config-prettier": "^1.18.0",
+    "typescript": "^3.6.3"
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/package.json]: Use latest typescript >= 3.7

File: examples/micro-http-server-ts/src/middlewares.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  const { a, b } = qs.parse(query);
+
+  store.set({ a, b });
+
+  logger.debug(`Persisted a: ${a}`);
+  logger.debug(`Persisted b: ${b}`);
+}
+
+/**
+ * Middleware to add the parameters `a` and `b` and set `sum` on the async store.
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/middlewares.ts]: Need one blank line after description (before params).

File: examples/micro-http-server-ts/src/service.ts (hunk 1 above)
Comment (sumansta) [reply examples/micro-http-server-ts/src/service.ts]: `store.getAll()` returns `TypeError: store.getAll is not a function`

Comment (cham11ng) [reply examples/micro-http-server-ts/src/service.ts]: looks like we've not released it yet.

Comment (mesaugat) [reply examples/micro-http-server-ts/src/service.ts]: @cham11ng We can release it, if it's not released yet.

File: examples/micro-http-server-ts/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  "description": "A basic example for async-store and micro",
+  "main": "index.js",
+  "scripts": {
+    "start": "backpack",
+    "build": "backpack build"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@leapfrogtechnology/async-store": "^1.0.2",
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/package.json]: Use latest release

File: examples/micro-http-server-ts/src/service.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      const data = store.getAll();
+      logger.info("Store contents: "+ JSON.stringify(data));
+      resolve();
+    }, 2000);
+  });
+}
+
+
+
+
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/service.ts]: .

File: examples/micro-http-server-ts/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import * as micro from "micro";
+import { IncomingMessage, ServerResponse } from "http";
+
+import * as store from "@leapfrogtechnology/async-store";
+
+import * as logger from "./logger";
+import { storeParams, calculateSum } from "./middlewares";
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/index.ts]: Single quotes please. Looks like you are using `prettier` or something.

File: examples/micro-http-server-ts/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+/**
+ * Handle incoming http request.
+ *
+ * @param {IncomingMessage} req
+ * @param {ServerResponse} res
+ */
+function handleRequest(req: IncomingMessage, res: ServerResponse) {
+  const a = store.get("a");
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/index.ts]: @sumansta Single quotes throughout the codebase.

File: examples/micro-http-server-ts/src/logger.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  return (store.getId() || '').substring(0, 8);
+}
+
+/**
+ * Print log message to the stdout / stderr.
+ *
+ * @param {string} level
+ * @param {string} message
+ */
+function log(level:string, message:string) {
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/logger.ts]:
```suggestion
function log(level: string, message: string) {
```

File: examples/micro-http-server-ts/src/logger.ts [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+  process.stdout.write(line);
+}
+
+/**
+ * Write info logs and associated request id to stdout.
+ *
+ * @param {string} message
+ */
+export function info(message:string) {
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/logger.ts]:
```suggestion
export function info(message: string) {
```

File: examples/micro-http-server-ts/src/logger.ts [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+  log('INFO', message);
+}
+
+/**
+ * Write debug logs and associated request id to stdout.
+ *
+ * @param {string} message
+ */
+export function debug(message:string) {
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/logger.ts]: Same

File: examples/micro-http-server-ts/src/logger.ts [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+  log('DEBUG', message);
+}
+
+/**
+ * Write error logs and associated request id to stdout.
+ *
+ * @param {any} err
+ */
+export function error(err:any) {
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/logger.ts]: Same.

File: examples/micro-http-server-ts/src/middlewares.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @param {IncomingMessage} req
+ * @param {ServerResponse} res
+ */
+export function calculateSum(req: IncomingMessage, res: ServerResponse) {
+  doSomethingAsync();
+
+  const a = +store.get("a");
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/middlewares.ts]: Single quotes.

File: examples/micro-http-server-ts/src/service.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+  logger.debug(`Persisted b: ${b}`);
+}
+
+/**
+ * An example function making use of the request context
+ * set in the store asynchronously (with delay).
+ */
+export function doSomethingAsync() {
+  // Do something with the request with a delay.
+  logger.debug("Simulating delayed access");
```
Comment (mesaugat) [inline examples/micro-http-server-ts/src/service.ts]: Single quotes.

File: examples/micro-http-server-ts/src/index.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+import * as store from "@leapfrogtechnology/async-store";
+
+import * as logger from "./logger";
+import { storeParams, calculateSum } from "./middlewares";
+
+const PORT = 3000;
+
+const server = micro((req:IncomingMessage, res:ServerResponse) => {
+  store.initialize()(() => {
+    storeParams((req.url || "").split("?", 2)[1]);
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/index.ts]: single quotes. Not sure why tslint didn't catch this.

File: examples/micro-http-server-ts/src/service.ts [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const data = store.getAll();
+      logger.info("Store contents: "+ JSON.stringify(data));
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/service.ts]: .

File: examples/micro-http-server-ts/src/logger.ts [hunk 19]
```diff
@@ -0,0 +1,57 @@
+import * as store from '@leapfrogtechnology/async-store';
+
+/**
+ * Get unique request id from store or return empty string.
+ *
+ * @returns {string}
+ */
+function getRequestId() {
+  return (store.getId() || '').substring(0, 8);
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/logger.ts]: The new version has a new method.
```suggestion
  return store.getShortId();
```

File: examples/micro-http-server-ts/src/service.ts [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+ * set in the store asynchronously (with delay).
+ */
+export function doSomethingAsync() {
+  // Do something with the request with a delay.
+  logger.debug('Simulating delayed access');
+
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const data = store.getAll();
+      logger.info('Store contents: '+ JSON.stringify(data));
```
Comment (kabirbaidhya) [inline examples/micro-http-server-ts/src/service.ts]:
```suggestion
      logger.info('Store contents: ' + JSON.stringify(data));
```

**Pull request merged**